test/batch/test_requests.py

```diff
@@ -25,23 +25,21 @@ def test_add_and_get_request_body(self, mock_get_valid_uuid):
         # invalid calls
         #######################################################################
         ## error messages
-        type_error_message = "All arguments must be of type string"
+        type_error_message_1 = "'from_object_class_name' argument must be of type str"
+        type_error_message_2 = "'from_property_name' argument must be of type str"
+        type_error_message_3 = "'to_object_class_name' argument must be of type str"
 
         with self.assertRaises(TypeError) as error:
             batch.add(10, "some_str", "some_str", "some_str")
-        check_error_message(self, error, type_error_message)
-
-        with self.assertRaises(TypeError) as error:
-            batch.add("some_str", batch, "some_str", "some_str")
-        check_error_message(self, error, type_error_message)
+        check_error_message(self, error, type_error_message_1)
 
         with self.assertRaises(TypeError) as error:
             batch.add("some_str", "some_str", True, "some_str")
-        check_error_message(self, error, type_error_message)
+        check_error_message(self, error, type_error_message_2)
 
         with self.assertRaises(TypeError) as error:
-            batch.add("some_str", "some_str", "some_str", 1.0)
-        check_error_message(self, error, type_error_message)
+            batch.add("some_str", "some_str", "some_str", "some_uuid", 1.0)
+        check_error_message(self, error, type_error_message_3)
```

Comment:
Could you replace this unit test with an integration test?

And there we could parameterize the test to check uuid + str